ompi/mca/coll/hcoll/coll_hcoll_dtypes.h

```diff
@@ -92,7 +92,19 @@ static dte_data_representation_t* ompi_datatype_2_dte_data_rep[OMPI_DATATYPE_MAX
 #else
     &DTE_ZERO,
 #endif
-    &DTE_ZERO                   /*OPAL_DATATYPE_UNAVAILABLE    25 */
+
+#if SIZEOF_LONG == 4
```

Comment:
You should use OMPI datatypes not OPAL.

Reply:
That doesn't seem to be how hcoll is using that array.  A little further down in the same file it has:
```
    int opal_type_id = dtype->super.id;
    ...
    dte_data_rep =  *ompi_datatype_2_dte_data_rep[opal_type_id];
```
so it seems to be an array mapping each opal type to something hcoll recognizes.  So if I'm adding another OPAL_DATATYPE_<something> entry I think I'd need to expand this array too.